crates/iota/src/name_commands.rs

```diff
@@ -336,37 +337,60 @@ impl NameCommand {
                 )
             }
             Self::GetUserData { domain, key } => {
-                let entry = get_registry_entry(&domain, context).await?;
+                let client = context.get_client().await?;
+                let mut res = client
+                    .http()
+                    .iota_names_lookup(&domain.to_string())
+                    .await?
+                    .ok_or_else(|| anyhow!("no record found for {domain}"))?
+                    .data;
 
                 if let Some(key) = key {
-                    let Some(value) = entry
-                        .name_record
-                        .data
-                        .contents
-                        .into_iter()
-                        .find(|entry| entry.key == key)
-                    else {
+                    if let Some((key, value)) = res.remove_entry(&key) {
+                        res = BTreeMap::new();
+                        res.insert(key, value);
+                    } else {
                         anyhow::bail!("no value found for key `{key}`");
-                    };
-                    NameCommandResult::UserData(VecMap {
-                        contents: vec![value],
-                    })
-                } else {
-                    NameCommandResult::UserData(entry.name_record.data)
+                    }
                 }
+                NameCommandResult::UserData(res)
             }
             Self::List { address } => {
-                let mut nfts = get_owned_nfts::<IotaNamesRegistration>(address, context).await?;
-                let subdomain_nfts =
-                    get_owned_nfts::<SubdomainRegistration>(address, context).await?;
-                nfts.extend(subdomain_nfts.into_iter().map(|nft| nft.into_inner()));
```

Comment:
How come this wasn't creating duplicates btw?

Reply:
Not sure what you mean

Reply:
Because they don't return the other one